koku/reporting/rate/models.py

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Models for identity and access management."""
+
+from uuid import uuid4
+
+from django.db import models
+
+
+class Rate(models.Model):
+    """A rate for calculating costs.
+
+    A rate is (price * metric_usage / timeunit).
```

Comment:
A rate should actually be just `price / unit`.

Some of the usage (like CPU) is currently reported in `cpu-core-hours`. So units are a little more complicated than just time. The rate is really price per thing we charge for.  So this rate would be `$ / cpu-core-hour` if using hours and USD. 

Some (like memory) is currently reported in GB. So here the unit would be `GB` and time would not factor in to the rate. This does raise the question of whether we want to or should be reporting memory usage in GB-hours as well.

Reply:
A rate always includes a time component. Some metrics have the time component built-in, like `cpu-core-hours`, which means we can set the `timeunit` field to `None` there. However, everything else is either a cost-over-time or a one-time charge.

As you say, GB-hours, not GB.

The model simply gives you the ability to specify whether it should be GB-hours or GB-days without needing to change the data collector. So, if all we're receiving is a raw count of units consumed, we can divide that count over the specified time interval.